services/src/unit-test/java/uk/gov/moj/sdt/services/cache/GlobalParametersCacheTest.java

```diff
@@ -30,142 +30,144 @@
  * $LastChangedBy: $ */
 package uk.gov.moj.sdt.services.cache;
 
-import org.easymock.EasyMock;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Test;
-
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.gov.moj.sdt.dao.api.IGenericDao;
 import uk.gov.moj.sdt.domain.GlobalParameter;
 import uk.gov.moj.sdt.domain.api.IGlobalParameter;
 import uk.gov.moj.sdt.services.mbeans.SdtManagementMBean;
 import uk.gov.moj.sdt.utils.AbstractSdtUnitTestBase;
 import uk.gov.moj.sdt.utils.mbeans.api.ISdtManagementMBean;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
 /**
  * Test class for global parameters cache.
  *
  * @author d130680
  */
-public class GlobalParametersCacheTest extends AbstractSdtUnitTestBase {
-
-    /**
-     * Global parameters cache.
-     */
-    private GlobalParametersCache cache;
+@ExtendWith(MockitoExtension.class)
+class GlobalParametersCacheTest extends AbstractSdtUnitTestBase {
 
     /**
-     * Management bean holding cache information.
+     * Generic dao.
      */
-    private ISdtManagementMBean managementMBean;
+    @Mock
+    private IGenericDao mockGenericDao;
 
     /**
-     * Generic dao.
+     * Global parameters cache.
      */
-    private IGenericDao mockGenericDao;
+    private GlobalParametersCache cache;
 
     /**
      * Dao query results.
      */
     private GlobalParameter[] result;
 
+    private static final String PARAM_1 = "param1";
+    private static final String PARAM_2 = "param2";
+    private static final String PARAM_3 = "param3";
+    private static final String PARAMETER_1 = "parameter 1";
+
+
     /**
      * Setup mock objects and inject DAO dependencies into our class.
      */
-    @Before
+    @BeforeEach
+    @Override
     public void setUp() {
         cache = new GlobalParametersCache();
-        mockGenericDao = EasyMock.createMock(IGenericDao.class);
         cache.setGenericDao(mockGenericDao);
-        managementMBean = new SdtManagementMBean();
+
+        ISdtManagementMBean managementMBean = new SdtManagementMBean();
         cache.setManagementMBean(managementMBean);

```

Comment:
Is it worth setting up a mocked SdtManagementBean for testing this class?

Reply:
ISdtManagementMBean managementMBean = Mockito.mock(ISdtManagementMBean.class);
created mock

Reply:
Does the expected behaviour also need to be set and verified for the mocked management bean? It's used by AbstractCacheControl.uncacheRequired() which is invoked from AbstractCacheControl.getValue() - all the tests in this class use getValue().

Reply:
verify is called

Reply:
Confirmed that SdtManagementMBean behaviour is verified by tests.